### DISCUSSION TITLE
Cannot view some identities

### DISCUSSION THREAD

**Discussion by tlull** [Keyfactor/ejbca-ce#48] [Q&A] [answered]
Hello,
my database contains some identities, where it is impossible to view the identitiy but it is possible to edit them.
When viewing the identity the GUI displays the following error message:

`Es ist eine 'Exception' aufgetreten.
java.lang.IllegalStateException: Failed to parse data map for certificate request history for '8f083dc4fbfe5b05591259032f438e88cf92f46b': org.xmlpull.v1.XmlPullParserException: expected START_TAG or END_TAG not TEXT (position: TEXT seen ...<void property="keyRecoverable"> \n <boolean>true</... @23:19) 
`

I have attached the exception stack trace from server.log.
The EJBCA version I'm running is "EJBCA 7.4.3.2 Community".

What is wrong with such identities and how can it be repaired?

Regards,
   Torsten
[server.log](https://github.com/Keyfactor/ejbca-ce/files/8954930/server.log)

**Comment by mike-agrenius-kushner** [comment, marked as answer]
Hi Torsten,

Wow, that is super weird. Well, the data in question got corrupted somehow – is this a database from a previous installation which has been upgraded? 

Fortunately the data in question isn't vital, it's stored in the Certificate Request History table as documented here: https://doc.primekey.com/ejbca/ejbca-operations/ejbca-ca-concept-guide/certificate-authority-overview/ca-fields#CAFields-UseCertificateRequestHistoryUse_Certificate_Request_History

We don't actually recommend using this feature anymore, and you should fairly safely be able to remove the data from that table being simply performing a 

delete from CertReqHistoryData; 

on your database, but naturally please make a backup copy before doing so in case I may be entirely wrong =)

Cheers,
Mike

**Comment by tlull** [comment]
Hello Mike,
thanks for your very fast answering.
Our EJBCA instance already has seen some upgrades. If I remember right we startet with version 3.10.3.

Where do I see the Certificate Request History within the GUI? Is this reflected by the "Identitiy -> Certificates View -> View Older/Newer"?
Which functionality do I lose if I delete CertReqHistoryData?

Regards,
   Torsten

**Reply by mike-agrenius-kushner** [reply]
Hi, 

Essentially nothing at all, this is mostly a vestigial function which was relevant about a decade ago =) 

Today you'd use the audit log to get that information.

Cheers,
Mike

**Comment by tlull** [comment]
Hello Mike,
I have cleared the table you have mentioned and the problem is soved in my testing environment. In addition I have deselected "Use Certificate Request History" in all CA definitions.
So the next step is to do this in production EJBCA.

Many thanks for your support and kind regards,
   Torsten

**Reply by mike-agrenius-kushner** [reply]
Hi Torsen,

Very pleased to hear that! 

Cheers,
Mike
